### PR TITLE
Add test case to validate mod operation

### LINKED ISSUE #313: Determine right code to generate for `%` operator on floats
Ballerina semantics for `%` for float are based on Java: https://ballerina.io/spec/lang/2021R1/#section_6.23

LLVM provides `frem` instruction: https://llvm.org/docs/LangRef.html#i-frem

Need to check whether these are the same and if not what to generate.

### PR DESCRIPTION
**Pull request by heshanpadmasiri**
## Purpose
Add test case to validate mod operation
Resolves #313

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

**Comment by jclark** [comment]
Does the test pass with jBallerina?

File: compiler/testSuite/FVfmod00.bal [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+function isNaN(float val) returns string {
+	if float:isNaN(val) {
+		return "true";
+	}
+	return "false";
+}
+
+function isEqual(float lhs, float rhs) returns string {
+    if lhs == rhs {
```
Comment (manuranga) [inline compiler/testSuite/FVfmod00.bal:142]: should we use `===` here to catch `-0.0` cases ?

**Comment by manuranga** [comment]
> Does the test pass with jBallerina?

Yes, (it gives methods tool large, but passes when broken up)

File: compiler/testSuite/FVfmod00.bal (hunk 1 above)
Comment (jclark) [reply compiler/testSuite/FVfmod00.bal:142]: Yes.

Comment (manuranga) [reply compiler/testSuite/FVfmod00.bal:142]: I'll fix when I am moving this form `F` to `V`.

**Pull request merged**